lib/eslint/eslint.js

```diff
@@ -522,6 +524,14 @@ class ESLint {
         return version;
     }
 
+    /**
+     * The default configuration that ESLint uses internally.
+     * @type {ConfigArray}
+     */
+    static get defaultConfig() {
+        return Object.freeze(defaultConfig);
```

Comment:
If we are going to freeze the default config, I think it's better to do that right away in the module where it is created rather than as a side effect of accessing it through this property. That way, we could see earlier if freezing the object breaks something.

Reply:
Makes sense, updated 👍🏻